feat: Cloud Native Buildpacks project descriptor manager (#30799)
renovatebot/renovate: lib/modules/manager/cnb/readme.md

The `cnb` manager updates Cloud Native Buildpacks project descriptor file (`project.toml`), which references builder and/or buildpack images by URIs.

> Updates are performed if the `project.toml` file is found and it conforms to the [project descriptor](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md) specification.

Reviewer: ```suggestion
Renovate can update a `project.toml` file if:

- It can find the file
- The file follows the [project descriptor](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md) specifications
```

> __Note__: buildpacks in the `io.buildpacks.group` array must be configured with the docker reference (`uri`) for this manager to work.

Reviewer: ```suggestion
If you use buildpacks in the `io.buildpacks.group` array, then you _must_ configure the Docker reference (uri`) for Renovate to work.
```

> ```toml

Reviewer: ```suggestion
```toml title="Example of a `project.toml` file with Docker reference URIs"
```

Let's add a title to the codeblock. [^1]

[^1]: [Material for MkDocs manual, adding a title to a codeblock](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-a-title)

> The `cnb` manager updates Cloud Native Buildpacks project descriptors in `project.toml` files.

Reviewer: @rarkins do you want to keep calling it `cnb`, or do you want to rename to `buildpack`?

Context:

> > I'm pretty sure the CNF now have a trademark in buildpacks so this will be the only valid usage of that name
> 
> I don't think that matters since the v2 buildpacks are in CNCF, aren't they?
> 
> But probably the name doesn't really matter anyway. The end user will not have any contact unless a `project.toml` is present and for internal use there is still the `readme.md` if in doubt.

Users will still see the `cnb` name in their Renovate config. To me it seems easier to have a readable name like `buildpack`, than an abbreviation `cnb`.

Reviewer 2: I would like it called `buildpacks` with an "s" at the end

Author: the manager has been renamed

A `project.toml` file can reference builder / buildpack images by URIs.
Renovate can update a `project.toml` file if:

- It can find the file

> - The file follows the [project descriptor](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md) specifications

Reviewer: ```suggestion
- The file follows the [project descriptor specifications](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md)
```

> - The file follows the [project descriptor specifications](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md)

Reviewer: You commited [ignore unsupported buildpack sources](https://github.com/renovatebot/renovate/pull/30799/commits/2b174a9b6933f2345809836b9650f35bfb15b21a).

Do we need to update the readme, so it explains the unsupported buildpack sources?

Author: done
